crowdsourcehinter/crowdsourcehinter.py

```diff
@@ -0,0 +1,352 @@
+    #
+    # Example: ["personal computer", "PC", "computerr"]
+    incorrect_answers = List([], scope=Scope.user_state)
+    # A dictionary of generic_hints. default hints will be shown to students when there are no matches with the
+    # student's incorrect answer within the hint_database dictionary (i.e. no students have made hints for the
+    # particular incorrect answer)
+    #
+    # Example: ["Make sure to check your answer for simple mistakes like typos!"]
+    generic_hints = List(default=[], scope=Scope.content)
+    # List of which hints have been shown to the student
+    # this list is used to prevent the same hint from showing up to a student (if they submit the same incorrect answers
+    # multiple times)
+    #
+    # Example: ["You misspelled computer, remove the last r."]
+    used = List([], scope=Scope.user_state)
+    # This is a dictionary of hints that have been reported. the values represent the incorrect answer submission, and the
+    # keys are the hints the corresponding hints. hints with identical text for differing answers will all not show up for the
+    # student.
+    #
+    # Example: {"desk": "You're completely wrong, the answer is supposed to be computer."}
+    reported_hints = Dict(default={}, scope=Scope.user_state_summary)
+    # This String represents the xblock element for which the hinter is running. It is necessary to manually
+    # set this value in the XML file under the format "hinting_element": "i4x://edX/DemoX/problem/Text_Input" .
+    # Setting the element in the XML file is critical for the hinter to work.
+    #
+    # TODO: probably should change the name from Element (problem_element? hinting_element?). Trying to
+    # just change the name didn't seem to operate properly, check carefully what is changed
+    Element = String(default="", scope=Scope.content)
+
+    def studio_view(self, context=None):
+        """
+        This function defines a view for editing the XBlock when embedding it in a course. It will allow
+        one to define, for example, which problem the hinter is for. It is unfinished and does not currently
+        work.
+        """
+        html = self.resource_string("static/html/crowdsourcehinterstudio.html")
+        frag = Fragment(html)
+        frag.add_javascript_url('//cdnjs.cloudflare.com/ajax/libs/mustache.js/0.8.1/mustache.min.js')
+        frag.add_css(self.resource_string("static/css/crowdsourcehinter.css"))
+        frag.add_javascript(self.resource_string("static/js/src/crowdsourcehinter_studio.js"))
+        frag.initialize_js('CrowdsourceHinterStudio', {'initial': str(self.initial_hints), 'generic': str(self.generic_hints), 'element': str(self.Element)})
+        return frag
+
+    @XBlock.json_handler
+    def set_initial_settings(self, data, suffix=''):
+        """
+        Set intial hints, generic hints, and problem element from the studio view.
+        """
+        initial = ast.literal_eval(str(data['initial_hints']))
```

Comment:
Minor: json would be better here, but I think Piotr has already made that change elsewhere.